components/SinglePageComponents/Comments.js

```diff
@@ -0,0 +1,213 @@
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+
+export default function Comments({ id }) {
+  const [name, setName] = useState("");
+  const [comment, setComment] = useState("");
+  const [comments, setComments] = useState([]);
+  const [rating, setRating] = useState(0);
+
+  useEffect(() => {
+    const comments = JSON.parse(localStorage.getItem(`comments-${id}`)) || [];
+    setComments(comments);
+  }, [id]);
+
+  const handleChange = (event) => {
+    if (event.target.name === "name") {
+      setName(event.target.value);
+    } else {
+      setComment(event.target.value);
+    }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newComment = { name, comment, rating };
+    const updatedComments = [...comments, newComment];
+    setComments(updatedComments);
+    localStorage.setItem(`comments-${id}`, JSON.stringify(updatedComments));
+    setName("");
+    setComment("");
+    setRating(0);
+  };
+
+  const handleDelete = (id) => {
+    const updatedComments = [...comments];
+    updatedComments.splice(id, 1);
+    setComments(updatedComments);
+    localStorage.setItem(`comments-${id}`, JSON.stringify(updatedComments));
+  };
+
+  const handleRating = (event) => {
+    setRating(event.target.value);
+  };
+
+  return (
+    <StyledCommentsContainer>
+      <StyledCommentsHeadline>Comments & Rating</StyledCommentsHeadline>
+      <StyledForm onSubmit={handleSubmit}>
+        <StyledNameInput
+          type="text"
+          placeholder="Name"
+          name="name"
+          value={name}
+          onChange={handleChange}
+        ></StyledNameInput>
+        <StyledRatingBox>
+          <input
+            type="radio"
+            name="rating"
+            value="1"
+            onClick={handleRating}
+            checked={rating === 1}
+          />
+          1
+          <input
+            type="radio"
+            name="rating"
+            value="2"
+            onClick={handleRating}
+            checked={rating === 2}
+          />
+          2
+          <input
+            type="radio"
+            name="rating"
+            value="3"
+            onClick={handleRating}
+            checked={rating === 3}
+          />
+          3
+          <input
+            type="radio"
+            name="rating"
+            value="4"
+            onClick={handleRating}
+            checked={rating === 4}
+          />
+          4
+          <input
+            type="radio"
+            name="rating"
+            value="5"
+            onClick={handleRating}
+            checked={rating === 5}
+          />
+          5
+        </StyledRatingBox>
+        <StyledSendButton type="submit">Send</StyledSendButton>
+        <StyledCommentInput
+          type="text"
+          placeholder="Comment"
+          name="comment"
+          value={comment}
+          onChange={handleChange}
+        ></StyledCommentInput>
+      </StyledForm>
+
+      <StyledComments>
```

Comment:
Check for `comments.length > 0` and paint this component only if there are comments.